File: 2418. Sort the People/solution.js
/**
 * @param {string[]} names
 * @param {number[]} heights
 * @return {string[]}
 */
 var sortPeople = function(names, heights) {
  let dictionary = {};
  for (let index = 0; index < names.length; index++){
    let name = names[index];
    let height = heights[index];
    dictionary[height] = name;
  }

  heights.sort(function(a, b){return b-a});

  let result = [];
  for (let index = 0; index < heights.length; index++){
    let height = heights[index];
    result.push(dictionary[height]);
  }

  return result;
};

File: 2418. Sort the People/solution.py
class Solution(object):
  def sortPeople(self, names, heights):
    """
    :type names: List[str]
    :type heights: List[int]
    :rtype: List[str]
    """
    dictionary = {}
    for name, height in zip(names, heights):
      dictionary[height] = name
      
    result = []
    for height in sorted(heights, reverse = True):
      result.append(dictionary[height])

    return result